WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PagesViewModel.kt

```diff
@@ -90,9 +86,13 @@ class PagesViewModel
     private val eventBusWrapper: EventBusWrapper,
     private val previewStateHelper: PreviewStateHelper,
     private val uploadStarter: UploadStarter,
+    private val pageListEventListenerFactory: PageListEventListener.Factory,
     @Named(UI_THREAD) private val uiDispatcher: CoroutineDispatcher,
     @Named(BG_THREAD) private val defaultDispatcher: CoroutineDispatcher
-) : ScopedViewModel(uiDispatcher) {
+) : ScopedViewModel(uiDispatcher), LifecycleOwner {
+    private val lifecycleRegistry = LifecycleRegistry(this)
```

Comment:
This creates a lifecycle, but we still need to manually change the states of the lifecycle by calling `lifecycleRegistry.markState(...)`. 

I think it might not be a best practice to implement Lifecycle in VMs - I understand you used the same approach as in PostList. It's up to you if you want to keep the `PageListEventListener` consistent with `PostListEventListener` and use LifeCycle in both or don't implement the Lifecycle and simply call `postListEventListener.onDestroy` from PagesViewModel manually.

Reply:
Okay! I am thinking to just call the `onDestroy` manually. Thanks for pointing this out. 🙏

Reply:
Resolved in 23c8df4

Reply:
👍 Yep I agree it's better. Good call. Thanks!